ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1306,6 +1308,122 @@ public void shouldFailTerminateOnInvalidDeleteTopicPattern() {
     ksqlResource.terminateCluster(request);
   }
 
+  @Test
+  public void shouldFailIfRegisterTopicAlreadyExists() {
+    // Given:
+    final String registerSql = "REGISTER TOPIC FOO WITH (kafka_topic='bar', value_format='json');";
+    metaStore.putTopic(new KsqlTopic("FOO", "foo", new KsqlJsonTopicSerDe(), true));
+
+    when(commandStore.enqueueCommand(any(), any(), any(), any()))
+        .thenThrow(new IllegalArgumentException("Should never get to the enqueue stage!"));
+
+    // the below (commented) statement should cause this test to pass without the direct
+    // topic put on the meta store, but it does not. FIXME(agavra)
```

Comment:
Not good to be checking in code with TODOs in it.   Either fix the underlying issue or, as a last resort, remove the todo. You can always raise a Github issue against yourself to track the outstanding task.

Reply:
I didn't want to couple functionality changes with refactors. I'll submit an issue and track it against myself.